### PR TITLE
Addon catalog UI

### PR DESCRIPTION
**Pull request by winkerVSbecks**
* AddonsHomeScreen
* AddonsCategoryScreen
* AddonsTagScreen
* Search UI components
* AddonsDetailScreen

### CODE REVIEW

File: src/components/basics/Cardinal.js [hunk 1]
```diff
@@ -12,19 +12,19 @@ const Count = styled.div`
   color: ${color.dark};
   display: block;
 
-  ${props =>
+  ${(props) =>
```
Comment (kylesuss) [inline src/components/basics/Cardinal.js:15]: Did the prettier config change? Curious why this happened a bunch

Comment (winkerVSbecks) [reply src/components/basics/Cardinal.js:15]: Prettier made this change a while back. Don't think it's a config thing. I use the VS code plugin. It respects the project level config.

Comment (kylesuss) [reply src/components/basics/Cardinal.js:15]: Oh yeah hmm. I just pulled down and got the same result. I guess we just haven't touched this file in a while ha.

File: site-metadata.js [hunk 2]
```diff
@@ -115,4 +115,14 @@ siteMetadata.urls.navLinks = [
...
 ];
 
+siteMetadata.urls.addonsLearnLinks = [
+  { title: 'How to install addons', icon: 'book', href: '' },
+  {
+    title: 'Create an addon',
+    icon: 'edit',
+    href: 'https://www.chromatic.com/blog/how-to-build-a-storybook-addon/',
+  },
+  { title: 'Add to the catalog', icon: 'add', href: '' },
```
Comment (kylesuss) [inline site-metadata.js:125]: No href?

File: src/components/layout/addons/AddonItem.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+const VerifiedBadge = styled(Icon).attrs({ icon: 'verified', block: true })`
+  color: ${(props) => (props.appearance === 'official' ? color.secondary : color.green)};
+  width: 14px;
+  height: 14px;
+  margin-left: ${spacing.padding.small}px;
+  margin-bottom: 2px;
+`;
+VerifiedBadge.propTypes = {
+  appearance: PropTypes.oneOf(['official', 'integrator']),
+};
```
Comment (kylesuss) [inline src/components/layout/addons/AddonItem.js:20]: Is there a benefit to adding 0 to a styled component, especially if they aren't required props?

File: site-metadata.js (hunk 2 above)
Comment (winkerVSbecks) [reply site-metadata.js:125]: Yea, these are placeholders for now. Just building out the UI first while we wait for the back-end and the docs to be updated.

File: src/components/layout/addons/AddonsLayout.js [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+  hideSidebar: PropTypes.bool,
+};
+
+const ToCContent = styled.div`
+  /* Hide ToC on mobile, the primary navigation is search */
+  display: none;
+
+  ${(props) =>
+    props.hideToC
+      ? css`
```
Comment (kylesuss) [inline src/components/layout/addons/AddonsLayout.js:70]: How are you deciding when to use `css` in these? This feels a bit out of place as I don't see it elsewhere.

File: src/components/layout/addons/AddonsList.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+const ListWrapper = styled.div`
+  > *:not(:last-child) {
+    margin-bottom: ${spacing.padding.medium}px;
+  }
+`;
+
+export const AddonsList = ({ addonItems, isLoading, ...props }) => {
+  const [visibleCount, setVisibleCount] = useState(6);
+  const items = addonItems.slice(0, visibleCount);
```
Comment (kylesuss) [inline src/components/layout/addons/AddonsList.js:17]: Could memoize this so it doesnt alter the array on each render.

**Review by kylesuss — APPROVED** [review]
Just looked at the code. Will leave the design to @domyen . Looks great! Only comments were nitpicky things or questions.

File: src/components/layout/addons/AddonItem.js (hunk 3 above)
Comment (winkerVSbecks) [reply src/components/layout/addons/AddonItem.js:20]: imo you get the same benefit as any other component. Was just doing it out of habit but, happy to adopt whatever convention we use in the codebase.

File: src/components/layout/addons/AddonsLayout.js (hunk 4 above)
Comment (winkerVSbecks) [reply src/components/layout/addons/AddonsLayout.js:70]: I was doing this blindly but, it seems like you only need to do this when interpolating a function. I'll push an update.

File: src/components/layout/addons/AddonItem.js (hunk 3 above)
Comment (kylesuss) [reply src/components/layout/addons/AddonItem.js:20]: Makes sense, though with other components when you have non-required props, you have to specify a default typically -- that's what I was getting at.

File: src/components/layout/addons/AddonsLayout.js (hunk 4 above)
Comment (kylesuss) [reply src/components/layout/addons/AddonsLayout.js]: Np sounds good. I know in previous iterations of the design system that Dom was consistently using css like this but I have noticed it seems to be the exception that you actually need it. Just trying to find common patterns for us.

File: src/components/layout/addons/AddonItem.js (hunk 3 above)
Comment (winkerVSbecks) [reply src/components/layout/addons/AddonItem.js:20]: Good point, these should actually be required

**Review by domyen — CHANGES_REQUESTED** [review]

File: src/components/layout/addons/AddonsLayout.js [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+
+  #addons-search {
+    font-size: ${typography.size.s2}px;
+    padding-left: 40px;
+    padding-top: 12px;
+    padding-bottom: 12px;
+  }
+
+  svg {
+    left: 16px;
```
Comment (domyen) [inline src/components/layout/addons/AddonsLayout.js:106]:
```suggestion
    left: 16px;
    font-size: ${typography.size.s2}px;
```

This tweaks makes the icon size larger

**Pull request merged**